From c3636d5c9fab0521de92fe0a31eb5413d2af102b Mon Sep 17 00:00:00 2001
From: Benoit Bovy <benbovy@gmail.com>
Date: Thu, 28 Nov 2024 11:08:40 +0100
Subject: [PATCH] cmake: exclude nanoarrow from all install targets

---
 CMakeLists.txt | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2ec4acb..5fb3e93 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -213,6 +213,9 @@ if(S2GEOGRAPHY_BUILD_TESTS)
 
   FetchContent_MakeAvailable(nanoarrow)
 
+  # do not install nanoarrow by default when running cmake --install
+  set_property(DIRECTORY ${nanoarrow_SOURCE_DIR} PROPERTY EXCLUDE_FROM_ALL YES)
+
   if(S2GEOGRAPHY_CODE_COVERAGE)
     add_library(coverage_config INTERFACE)
   endif()
-- 
2.36.0

